apps/youtube_qa/multi_youtube_video_qa/multi_youtube_video_qa.py

```diff
@@ -0,0 +1,324 @@
+        else:
+            raw_transcript_string = generate_online_video_transcript(cursor)           
+            partitioned_transcript = partition_transcript(raw_transcript_string)
+            df = pd.DataFrame(partitioned_transcript)
+            print(df)
+            if os.path.exists("transcript.csv"):
+                df.to_csv("transcript.csv", mode='a')
+            else:
+                df.to_csv("transcript.csv")
+
+        print("Total transcription time : ", total_transcription_time)
+
+        load_start_time = time.time()
+        # load chunked transcript into table
+        cursor.query("""DROP TABLE IF EXISTS Transcript;""").execute()
+        cursor.query(
+            """CREATE TABLE IF NOT EXISTS Transcript (text TEXT(100));"""
+        ).execute()
+        cursor.load("transcript.csv", "Transcript", "csv").execute()
+        print(f"Loading transcripts into DB took {time.time() - load_start_time} seconds")
+        
+        print("Creating embeddings and Vector Index")
+        
+        udf_query = """CREATE UDF IF NOT EXISTS embedding
+            IMPL  '../../../evadb/udfs/sentence_feature_extractor.py';
+            """
+        cursor.query("DROP UDF IF EXISTS embedding;").execute()
+        cursor.query(udf_query).execute()
+
+        cursor.query("DROP UDF IF EXISTS ChatGPT;").execute()
+        cursor.query(""" CREATE UDF IF NOT EXISTS ChatGPT
+          IMPL '../../../evadb/udfs/chatgpt.py';""").execute()
+
+        cursor.drop_table("embedding_table").execute()
+        est = time.time()
+        cursor.query(
+            f"""CREATE TABLE embedding_table AS
+            SELECT embedding(text), text FROM Transcript;"""
+        ).execute()
+        eft = time.time()
+        print(f"Creating embeddings took {eft - est} seconds")
+
+        # Create search index on extracted features.
+        cursor.query(
+            f"CREATE INDEX faiss_index ON embedding_table (features) USING FAISS;"
+        ).execute()
+
+        vet = time.time()
+        print(f"Creating index took {vet - eft} seconds")
+
+        questions = open('questions', 'r')
+        st = time.time()
+        for question in questions:
+            print(question)
+            question_start = time.time()
+            
+            # instead of passing all the documents to the LLM, we first do a
+            # semantic search over the embeddings and get the most relevant rows.
+            cursor.query("DROP TABLE IF EXISTS EMBED_TEXT;").execute()
+            text_summarization_query = f"""
+                CREATE MATERIALIZED VIEW 
+                EMBED_TEXT(text) AS 
+                SELECT text FROM embedding_table
+                ORDER BY Similarity(embedding('{question}'), features) DESC
+                LIMIT 3; 
+                """
+            cursor.query(text_summarization_query).execute()
+            print(cursor.table('EMBED_TEXT').select("*").execute())
+
+            chatgpt_udf = """
```

Comment:
we should use api instead of explicit query. for instance, cursor.table(EMBED_TEXT).select(f"ChatGPT('{question}', text)")

Reply:
Had used API only. Missed removing this declaration.